client/my-sites/plans-v2/upsell.tsx

```diff
@@ -1,41 +1,187 @@
+						translate( 'Would you like to add %s?', {
+							args: [ upsellProductName ],
+							comment: '%s is the name of a product such as Jetpack Scan or Jetpack Backup',
+						} )
+					) }
+					brandFont
+				/>
+				<div className="upsell__icons">
+					<ProductIcon className="upsell__product-icon" slug={ mainProduct.iconSlug } />
+					<Gridicon className="upsell__plus-icon" icon="plus-small" />
+					<ProductIcon className="upsell__product-icon" slug={ upsellProduct.iconSlug } />
+				</div>
+				<p className="upsell__subheader">
+					{ preventWidows(
+						translate( 'Bundle %(mainProduct)s with %(upsellProduct)s and save!', {
+							args: {
+								mainProduct: mainProductName,
+								upsellProduct: upsellProductName,
+							},
+							comment: '%s are abbreviated name of product such as Scan or Backup',
+						} )
+					) }
+				</p>
+				<p className="upsell__saved-amount">
+					{ preventWidows(
+						translate(
+							'Save $%(savedAmount)d/year on %(upsellProduct)s when paired with %(mainProduct)s',
+							{
+								args: {
+									savedAmount,
+									mainProduct: mainProductName,
+									upsellProduct: upsellProductName,
+								},
+							}
+						)
+					) }
+				</p>
+			</div>
+
+			<JetpackProductCard
+				iconSlug={ upsellProduct.iconSlug }
+				productName={ upsellProduct.displayName }
+				subheadline={ upsellProduct.tagline }
+				description={ upsellProduct.description }
+				currencyCode={ currencyCode }
+				billingTimeFrame={ durationToText( upsellProduct.term ) }
+				buttonLabel={ translate( 'Yes, add %s', { args: [ upsellProductName ] } ) }
+				features={ { items: [] } }
+				discountedPrice={ upsellProductPrices.discountCost }
+				originalPrice={ upsellProductPrices.cost || 0 }
+				onButtonClick={ onPurchaseBothProducts }
+				cancelLabel={ translate( 'No, I do not want %s', {
+					args: [ upsellProductName ],
+				} ) }
+				onCancelClick={ onPurchaseSingleProduct }
+			/>
+		</Main>
+	);
+};
 
 const UpsellPage = ( { duration, productSlug, rootUrl }: UpsellPageProps ) => {
-	const siteSlug = useSelector( ( state ) => getSelectedSiteSlug( state ) ) || '';
+	const selectedSiteSlug = useSelector( ( state ) => getSelectedSiteSlug( state ) ) || '';
+	const isFetchingProducts = useSelector( ( state ) => isProductsListFetching( state ) );
+	const currencyCode = useSelector( ( state ) => getCurrentUserCurrencyCode( state ) );
+
+	const mainProduct = slugToSelectorProduct( productSlug );
 	// TODO: Get upsells via API.
+	const upsellProductSlug = getProductUpsell( productSlug );
+	const upsellProduct = upsellProductSlug && slugToSelectorProduct( upsellProductSlug );
 
-	// If the product is not valid, send the user to the selector page.
-	const product = slugToSelectorProduct( productSlug );
-	if ( ! product ) {
+	// If the product is not valid or there is no upsell product for it,
+	// send the user to the selector page.
+	if ( ! mainProduct || ! upsellProduct ) {
```

Comment:
Perhaps the code could instead send the user to the checkout page if there's no upsell product? The current structure of does not check if an upsell exists before sending the user to this page.

Reply:
This a very good idea.